common/src/main/java/org/apache/uniffle/common/netty/protocol/GetLocalShuffleDataResponse.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.uniffle.common.netty.protocol;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+
+import org.apache.uniffle.common.rpc.StatusCode;
+import org.apache.uniffle.common.util.ByteBufUtils;
+
+public class GetLocalShuffleDataResponse extends RpcResponse {
+  private ByteBuf data;
+
+  public GetLocalShuffleDataResponse(long requestId, StatusCode statusCode, byte[] data) {
+    this(requestId, statusCode, null, data);
+  }
+
+  public GetLocalShuffleDataResponse(long requestId, StatusCode statusCode, String retMessage, byte[] data) {
+    this(requestId, statusCode, retMessage, Unpooled.wrappedBuffer(data));
+  }
+
+  public GetLocalShuffleDataResponse(long requestId, StatusCode statusCode, String retMessage, ByteBuf data) {
+    super(requestId, statusCode, retMessage);
+    this.data = data;
+  }
+
+  @Override
+  public int encodedLength() {
+    return super.encodedLength() + Integer.BYTES + data.readableBytes();
+  }
+
+  @Override
+  public void encode(ByteBuf buf) {
+    super.encode(buf);
+    ByteBufUtils.copyByteBuf(data, buf);
```

Comment:
Could we reduce this copy? Do we need release this ByteBuf?

Reply:
Not yet, this is a todo
We discussed this issue in #826, in fact, the following two conditions need to be met to reduce this copy
1. Implement a MessageWithHeader
2. The above data is used as `body`

In this way, the data of data can be directly copied to`WritableByteChannel`, but there are some problems in the current implementation of point 2. For example, our SendShuffleData contains multiple blocks, and some modifications needs to be done, such as the metadata in ShuffleBlockInfo and binary data separated, In order to treat the binary data of all blocks as a `body`.

Reply:
Sure, we need to release, i will fix.

Reply:
`MessageWithHeader ` is introduced to support zero-copy, See https://github.com/apache/spark/blob/master/common/network-common/src/main/java/org/apache/spark/network/protocol/MessageWithHeader.java.

Reply:
cc @smallzhongfeng